### PR TITLE
[Eval] Get logits output.

### PR DESCRIPTION
**Pull request by marvin-Yu**

### CODE REVIEW

File: include/models.h [hunk 1]
```diff
@@ -70,6 +76,7 @@ class Model {
     std::vector<int32_t> inputIds;
     int batchSize;
     int seqLen;
+    int vocabSize_;
```
Comment (pujiang2018) [inline include/models.h:79]: Why do we need vocabSize in Model class?
if needed, could you pls make it in same naming style?

File: src/pytorch/auto_model.h [hunk 2]
```diff
@@ -129,6 +129,25 @@ struct TorchAutoModel : torch::CustomClassHolder {
                 doSample, temperature, topK, topP, repetitionPenalty, stopWordsList_int32);
     }
 
+    torch::Tensor forward(torch::Tensor &inputIds) {
+        int batchSize = inputIds.size(0);
+        int seqLen = inputIds.size(1);
+        int vocabSize = model->getVocabSize();
+        int logitsN = batchSize * seqLen * vocabSize;
+
+        if (model->getRank() == 0) { input(inputIds); }
```
Comment (pujiang2018) [inline src/pytorch/auto_model.h:138]: For the original code, why 'input' method is not called 'setInput', input is more like a keyword, :(

**Comment by pujiang2018** [comment]
overall LGTM. @Duyi-Wang please help to review as it is closely related with the interface.

File: include/models.h (hunk 1 above)
Comment (marvin-Yu) [reply include/models.h]: Done

**Comment by Duyi-Wang** [comment]
I think it's better to keep align with `transformers` to use `output_scores=True` in `generate()` instead of `forward()`.

output_scores (bool, optional, defaults to False) — Whether or not to return the prediction scores. See scores under returned tensors for more details.
output_logits (bool, optional) — Whether or not to return the unprocessed prediction logit scores. See logits under returned tensors for more details.
https://huggingface.co/docs/transformers/main_classes/text_generation#transformers.GenerationConfig.output_scores

File: src/pytorch/auto_model.h [hunk 3]
```diff
@@ -129,6 +129,25 @@ struct TorchAutoModel : torch::CustomClassHolder {
...
+
+        std::tuple<float *, int, int> result = model->forward();
+        float *outBuf = std::get<0>(result);
+        int sampleOffset = std::get<1>(result);
+        int sampleSize = std::get<2>(result);
+
+        // Create a torch::Tensor from the C array
+        int64_t tdims[3] = {batchSize, seqLen, vocabSize};
```
Comment (Duyi-Wang) [inline src/pytorch/auto_model.h:146]: Shape is correct? The decoder just returns the last token's logits.

File: src/pytorch/auto_model.h [hunk 4]
```diff
@@ -129,6 +129,25 @@ struct TorchAutoModel : torch::CustomClassHolder {
...
+        int seqLen = inputIds.size(1);
+        int vocabSize = model->getVocabSize();
+        int logitsN = batchSize * seqLen * vocabSize;
+
+        if (model->getRank() == 0) { input(inputIds); }
+
+        std::tuple<float *, int, int> result = model->forward();
+        float *outBuf = std::get<0>(result);
+        int sampleOffset = std::get<1>(result);
+        int sampleSize = std::get<2>(result);
```
Comment (Duyi-Wang) [inline src/pytorch/auto_model.h:143]: Sync in multi-ranks?

File: src/pytorch/auto_model.h (hunk 3 above)
Comment (marvin-Yu) [reply src/pytorch/auto_model.h:146]: It opens the "logitsAll" to true for all logit output.
```
decoder->forward(..., logitsAll = true);
```

**Comment by marvin-Yu** [comment]
> I think it's better to keep align with `transformers` to use `output_scores=True` in `generate()` instead of `forward()`.
> 
> output_scores (bool, optional, defaults to False) — Whether or not to return the prediction scores. See scores under returned tensors for more details. output_logits (bool, optional) — Whether or not to return the unprocessed prediction logit scores. See logits under returned tensors for more details. https://huggingface.co/docs/transformers/main_classes/text_generation#transformers.GenerationConfig.output_scores

This implementation aligns with the approach on Hugging Face, which is merely a model's execution from input (token id) to output (logits), without considering the searcher component.

File: src/pytorch/auto_model.h (hunk 4 above)
Comment (marvin-Yu) [reply src/pytorch/auto_model.h:143]: not support in this PR.

**Review by Duyi-Wang — APPROVED** [review]
